great_expectations/data_context/types/base.py

```diff
@@ -2239,8 +2240,27 @@ class CheckpointValidationConfigSchema(Schema):
     class Meta:
         unknown = INCLUDE
 
-    name = fields.String(required=False, allow_none=False)
-    id_ = fields.String(required=False, allow_none=False)
+    id_ = fields.String(required=False, allow_none=False, data_key="id")
+
+    def dump(self, obj: dict, *, many: Optional[bool] = None) -> dict:
```

Comment:
Do we have an explicit list of what can go in validations nested within checkpoints? If so, we can get rid of this. As this has historically been a dictionary, I wanted to guard against edge cases and make this as forgiving of unknown values as possible.

Reply:
I did not find a list in a quick search, but what about using all of the keys in validations from https://docs.greatexpectations.io/docs/guides/validation/checkpoints/how_to_configure_a_new_checkpoint_using_test_yaml_config and https://docs.greatexpectations.io/docs/guides/validation/checkpoints/how_to_add_validations_data_or_suites_to_a_checkpoint as a starting point?

Reply:
Also take a look at get_substituted_validation_dict().
Maybe best to create the list but also leave it open until we can make sure we have them all.